src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile_isolated_strategy.py

```diff
@@ -38,6 +40,12 @@ def __init__(self, context, options, workdir, analysis_tools, language, sources_
     self._analysis_dir = os.path.join(workdir, 'isolated-analysis')
     self._classes_dir = os.path.join(workdir, 'isolated-classes')
 
+
+    # A temporary, but well-known, dir in which to munge analysis/dependency files in before
+    # caching. It must be well-known so we know where to find the files when we retrieve them from
+    # the cache.
+    self._analysis_tmpdir = os.path.join(self._analysis_dir, 'artifact_cache_tmpdir')
```

Comment:
You could just get rid of this `_` variable all together by moving `os.path.join(...)` into the `def analysis_tmpdir` below.  If you don't want to compute it more than once, you can mark it with `@memoized_property`   See `src/python/pants/util/memo.py`